### PR TITLE
first pass at drop

### PR DESCRIPTION
**Pull request by chrisnojima**
first drop at the initial drop stuff
the design will change
integration hasn't happened yet
but lets try and get this in so we can work on it
its all behind a feature flag

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers

**Comment by chrisnojima** [comment]
actually back to WIP as things are moving around a bunch

**Comment by chrisnojima** [comment]
- [ ] bring back hide gregor action
- [ ] update assets
- [ ] update progress

**Comment by chrisnojima** [comment]
@cecileboucheron updated

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

**Comment by chrisnojima** [comment]
i made this opt in so override the feature flag for the drop

**Comment by chrisnojima** [comment]
@keybase/react-hackers can i get another review.

**Comment by chrisnojima** [comment]
@buoyad @MarcoPolo updated

File: shared/wallets/airdrop/qualify/index.js [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    if (!shallowEqual(this.props.rows, prevProps.rows)) {
+      this.setState({rowIdxLoaded: -1})
+      this._kickNextLoad()
+    } else if (this.state.rowIdxLoaded < this.props.rows.length) {
+      this._kickNextLoad()
+    }
+  }
+
+  render() {
+    const p = this.props
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/wallets/airdrop/qualify/index.js]: this is just cause it went from a function to a class, we do this all over the place

File: shared/actions/wallets.js [hunk 2]
```diff
@@ -900,6 +901,85 @@ const exitFailedPayment = (state, action) => {
...
+    section?: ?string,
+    lines?: ?Array<?{
+      bullet?: ?boolean,
+      text?: ?string,
+    }>,
+  }>,
+}
+const updateAirdropDetails = () =>
+  RPCStellarTypes.localAirdropDetailsLocalRpcPromise(undefined, Constants.airdropWaitingKey).then(s => {
+    const json: AirdropDetailsJSONType = JSON.parse(s)
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/actions/wallets.js]: cause it was just simpler to make it json. i discussed w/ patrick, its fine for now

File: shared/app/nav.desktop.js [hunk 3]
```diff
@@ -46,6 +47,7 @@ class Nav extends React.Component<Props> {
           )}
           <ErrorBoundary>
             <Box style={{...globalStyles.flexBoxColumn, flex: 1}}>
+              {routeSelected !== Tabs.loginTab && routeSelected !== Tabs.walletsTab && <AirdropBanner />}
```
[the remark this answers is not in the document]
Comment (chrisnojima) [reply shared/app/nav.desktop.js]: thats ok

**Comment by chrisnojima** [comment]
- [ ] after release

File: shared/wallets/airdrop/qualify/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/wallets/airdrop/qualify/index.js]: Ah ok, I'm used to it being `props`

File: shared/common-adapters/popup-dialog.desktop.js [hunk 4]
```diff
@@ -62,6 +62,7 @@ const styles = Styles.styleSheetCreate({
       borderRadius: Styles.borderRadius,
       flex: 1,
       maxWidth: '100%',
+      overflow: 'hidden',
```
[the remark this answers is not in the document]
Comment (buoyad) [reply shared/common-adapters/popup-dialog.desktop.js]: Ditto

File: shared/wallets/wallet-list/index.js [hunk 5]
```diff
@@ -102,9 +131,13 @@ class WalletList extends React.Component<Props> {
         </Kb.Box2>
       )
     }
+
     const rows = this.props.accountIDs.map(accountID => ({accountID, key: accountID, type: 'wallet'}))
+
     const addWallet = 'add wallet'
     rows.push({key: addWallet, type: addWallet})
+    const joinAirdrop = 'join airdrop'
+    rows.push({key: joinAirdrop, type: joinAirdrop})
```
Comment (buoyad) [inline shared/wallets/wallet-list/index.js]: Disable if `!flags.airdrop`

File: shared/tracker2/bio/index.js [hunk 6]
```diff
@@ -12,13 +12,27 @@ type Props = {|
...
+      <Kb.Text
+        center={true}
+        type="BodyBig"
+        lineClamp={p.inTracker ? 1 : undefined}
+        style={styles.text}
+        selectable={true}
+      >
+        {p.fullname}
+      </Kb.Text>
+      {p.registeredForAirdrop && (
```
Comment (buoyad) [inline shared/tracker2/bio/index.js]: Disable if `!flags.airdrop`, also seems like the `centered` -> `center` change messed up the alignment.
image

**Comment by chrisnojima** [comment]
@buoyad updated

**Comment by chrisnojima** [comment]
- [ ] after release

**Pull request merged**